From fdb532ff7851f07a77090636c5b162aebcfdae8a Mon Sep 17 00:00:00 2001
From: Dmitry Muhomor <muhomor.dmitry@gmail.com>
Date: Wed, 8 Oct 2025 22:50:15 +0300
Subject: [PATCH] gmscompat: remove error string from doDump()

GmsCore calls doDump() with "--displays" arg and fails if "Permission Denial..." is returned.
It's not clear what's the purpose of that check, but returning nothing makes it pass.
---
 services/surfaceflinger/SurfaceFlinger.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index 7565080d2..3f47a1941 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -6643,9 +6643,10 @@ status_t SurfaceFlinger::doDump(int fd, const DumpArgs& args, bool asProto) {
     const int uid = ipc->getCallingUid();
 
     if ((uid != AID_SHELL) && !PermissionCache::checkPermission(sDump, pid, uid)) {
-        StringAppendF(&result, "Permission Denial: can't dump SurfaceFlinger from pid=%d, uid=%d\n",
-                      pid, uid);
-        write(fd, result.c_str(), result.size());
+        ALOGD("doDump: callingPid %i, callingUid %i, asProto %i", pid, uid, (int) asProto);
+        for (auto &arg : args) {
+            ALOGD("doDump: arg %s", String8(arg).c_str());
+        }
         return NO_ERROR;
     }
 
-- 
2.43.0

